tests/cookbook/test_write.py

```diff
@@ -24,12 +28,92 @@ def test_parquet_write_with_partitioning(tmp_path):
     pd_df = df.write_parquet(tmp_path, partition_cols=["Borough"])
 
     read_back_pd_df = daft.read_parquet(tmp_path.as_posix() + "/**/*.parquet").to_pandas()
-    assert_df_equals(df.exclude("Borough").to_pandas(), read_back_pd_df)
+    assert_df_equals(df.to_pandas(), read_back_pd_df)
 
     assert len(pd_df) == 5
     assert len(pd_df._preview.preview_partition) == 5
 
 
+def test_empty_parquet_write_without_partitioning(tmp_path):
+    df = daft.read_csv(COOKBOOK_DATA_CSV)
+    df = df.where(daft.lit(False))
+    output_files = df.write_parquet(tmp_path)
+    assert len(output_files) == 0
+    assert len(output_files._preview.preview_partition) == 0
+
+
+def test_empty_parquet_write_with_partitioning(tmp_path):
+    df = daft.read_csv(COOKBOOK_DATA_CSV)
+    df = df.where(daft.lit(False))
+    output_files = df.write_parquet(tmp_path, partition_cols=["Borough"])
+    assert len(output_files) == 0
+    assert len(output_files._preview.preview_partition) == 0
+
+
+def test_parquet_write_with_partitioning_readback_values(tmp_path):
+    df = daft.read_csv(COOKBOOK_DATA_CSV)
+
+    output_files = df.write_parquet(tmp_path, partition_cols=["Borough"])
+    output_dict = output_files.to_pydict()
+    boroughs = {"QUEENS", "MANHATTAN", "STATEN ISLAND", "BROOKLYN", "BRONX"}
+    assert set(output_dict["Borough"]) == boroughs
+
+    for path, bor in zip(output_dict["path"], output_dict["Borough"]):
+        assert f"Borough={bor}" in path
+        read_back = daft.read_parquet(path).to_pydict()
+        assert all(b == bor for b in read_back["Borough"])
+
+    read_back_pd_df = daft.read_parquet(tmp_path.as_posix() + "/**/*.parquet").to_pandas()
+    assert_df_equals(df.to_pandas(), read_back_pd_df)
+
+    assert len(output_files) == 5
+    assert len(output_files._preview.preview_partition) == 5
+
+
+@pytest.mark.skipif(
+    not PYARROW_GE_7_0_0,
+    reason="We only use pyarrow datasets 7 for this test",
+)
+def test_parquet_write_with_null_values(tmp_path):
+    df = daft.from_pydict({"x": [1, 2, 3, None]})
+    df.write_parquet(tmp_path, partition_cols=[df["x"].alias("y")])
+    ds = pads.dataset(tmp_path, format="parquet", partitioning=pads.HivePartitioning(pa.schema([("y", pa.int64())])))
+    readback = ds.to_table()
+    assert readback.to_pydict() == {"x": [1, 2, 3, None], "y": [1, 2, 3, None]}
+
+
+@pytest.mark.skipif(
+    not PYARROW_GE_7_0_0,
+    reason="We only use pyarrow datasets 7 for this test",
+)
+def test_parquet_write_multifile(tmp_path):
+    daft.set_execution_config(parquet_target_filesize=1024)
```

Comment:
We use a fixture in other tests to ensure that the old configs are restored after this test runs

```
try:
    ... # set new configs
    yield
finally:
    # restore old configs
```

Reply:
+1 to this! Here's an example:
- https://github.com/Eventual-Inc/Daft/blob/e843bc0fb715a910448efda73b0215b26e3ea99e/tests/dataframe/test_joins.py#L12-L24
- https://github.com/Eventual-Inc/Daft/blob/e843bc0fb715a910448efda73b0215b26e3ea99e/tests/dataframe/test_joins.py#L27-L47

Reply:
done!